build-profiling-ffi.sh

```diff
@@ -165,7 +165,7 @@ FEATURES=$(IFS=, ; echo "${FEATURES[*]}")
 echo "Building for features: $FEATURES"
 
 # build inside the crate to use the config.toml file
-( cd datadog-profiling-ffi && DESTDIR="$destdir" cargo rustc --features $FEATURES --release --target "${target}" --crate-type cdylib && DESTDIR="$destdir" cargo rustc --features $FEATURES --release --target "${target}" --crate-type staticlib)
+( cd datadog-profiling-ffi && DESTDIR="$destdir" cargo rustc --features $FEATURES --release --target "${target}" --crate-type cdylib -- -C opt-level="s" && DESTDIR="$destdir" cargo rustc --features $FEATURES --release --target "${target}" --crate-type staticlib -- -C opt-level="s")
```

Comment:
Would we not prefer having this within the `[profile.release]` of the `./datadog-profiling-ffi/.cargo/config.toml` file ?

The general direction was to remove most of what is put within the shell script to ensure it is really described within the build system itself.

Reply:
actually this needs to go draft again, it seems we have hit an issue with v18 built process